src/net_processing.cpp

```diff
@@ -3567,6 +3601,8 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             // Don't use timedata samples from inbound peers to make it
             // harder for others to tamper with our adjusted time.
             AddTimeData(pfrom.addr, peer->m_time_offset);
+
+            m_outbound_time_offsets.Add(peer->m_time_offset);
```

Comment:
Having a rolling set of 10 time offsets from every outbound connection we make (full outbounds, block-relay-only, extra block-relay-only, feelers?) seems somewhat risky: at present, you'll get odd behaviour if a high-proportion of the first 100 outbound peers you connect to have a different time; with this change, you'll get a warning if the last 10 peers you randomly connect to have a dodgy view of the time?

Maybe consider making this conditional on `IsManualOrFullOutboundConn()` instead; that way it only updates rarely for things we're really trying to connect to, and isn't mostly made up of random peers we're briefly sampling?